### PR TITLE
Modal bugfixes

### PR DESCRIPTION
**Pull request by mennoo1996**
Fixed some bugs in new/edit project.
Add edit buttons for camera and camera type
Gave functionality to cancel buttons
There is now a warning prompt when a camera type is deleted that is used. If the user confirms, the cameras are automatically deleted as well

Such a prompt should also be built for deleting cameras, but for this I first need more insight in how DirectorBlocks and CameraBlocks are linked.

### CODE REVIEW

File: src/main/java/control/ProjectController.java [hunk 1]
```diff
@@ -303,6 +342,141 @@ public void newProject() {
...
+                            editProjectModal.getCameraTypes().get(selectedIndex))) {
+                        e.setCameraType(type);
+                    }
+                });
+            
+            editProjectModal.getCameraTypes().set(selectedIndex, type);
+            HBox box = new HBox();
+            box.getChildren().addAll(new Label(name), new Label(" - "), new Label(description));
+            editProjectModal.getCameraTypeList().getItems().set(selectedIndex, box);
+
```
Comment (bart-de-jonge) [inline src/main/java/control/ProjectController.java]: unnecessary whitespace

File: src/main/java/control/ProjectController.java [hunk 2]
```diff
@@ -303,6 +342,141 @@ public void newProject() {
...
+    
+    /**
+     * Event handler for when the cancel button in the camera type edit modal is clicked.
+     * @param event the MouseEvent for this event
+     */
+    private void cameraTypeEditCancelled(MouseEvent event) {
+        cameraTypeModal.hideModal();
+    }
+    
+
```
Comment (bart-de-jonge) [inline src/main/java/control/ProjectController.java]: whitespace

File: src/main/java/gui/modal/AddCameraModalView.java [hunk 3]
```diff
@@ -75,6 +75,26 @@ public AddCameraModalView(RootPane rootPane,
...
+        this(rootPane, types, width, height);
+        this.nameField.setText(camera.getName());
+        this.descriptionField.setText(camera.getDescription());
+        this.cameraTypes.getSelectionModel().select(typeIndex);
+        this.addCameraButton.setText("Save");
+        this.titleLabel.setText("Edit a camera");
+        
+    }
+    
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/modal/AddCameraModalView.java]: whitespace

File: src/main/java/gui/root/RootCenterArea.java [hunk 4]
```diff
@@ -148,13 +148,16 @@ private void initTopPane() {
...
                     .getScriptingProject().getCameraTimelines().get(i).getName();
-            Label cameraLabel = new Label(name);
-            cameraLabel.prefWidthProperty().bind(getMainTimeLineGridPane()
-                    .widthProperty().divide(numberOfTimelines));
-            cameraLabel.setAlignment(Pos.CENTER);
-            cameraLabel.setPadding(new Insets(topBarHeight / 2.0, 0,
+
+            Label label = new Label(name);
+            
+
```
Comment (bart-de-jonge) [inline src/main/java/gui/root/RootCenterArea.java]: whitespace

**Comment by bart-de-jonge** [comment]
Some small comments about whitespace, other than that it looks fine

File: src/main/java/data/Camera.java [hunk 5]
```diff
@@ -55,7 +55,18 @@ public Camera(String name, String description, CameraType cameraType) {
         log.debug("Created new Camera(name={}, description={}, cameraType={}",
                 name, description, cameraType);
     }
-
+    
+    /**
+     * Clone constructor.
+     * @param camera camera to clone.
+     */
+    public Camera(Camera camera) {
```
Comment (geenen124) [inline src/main/java/data/Camera.java]: Take a look at the [Cloneable](https://docs.oracle.com/javase/7/docs/api/java/lang/Cloneable.html) interface

File: src/main/java/data/CameraTimeline.java [hunk 6]
```diff
@@ -60,6 +60,18 @@ public CameraTimeline(Camera camera, String description, ScriptingProject projec
     }
     
     /**
+     * Clone constructor.
+     * @param timeline timeline to clone
+     */
+    public CameraTimeline(CameraTimeline timeline) {
```
Comment (geenen124) [inline src/main/java/data/CameraTimeline.java]: [Cloneable](https://docs.oracle.com/javase/7/docs/api/java/lang/Cloneable.html)

File: src/main/java/data/CameraType.java [hunk 7]
```diff
@@ -37,6 +37,16 @@
     public CameraType() {
         this("", "", -1);
     }
+    
+    /**
+     * Clone constructor.
+     * @param type type to clone
+     */
+    public CameraType(CameraType type) {
```
Comment (geenen124) [inline src/main/java/data/CameraType.java]: [Cloneable](https://docs.oracle.com/javase/7/docs/api/java/lang/Cloneable.html)
